src/Cli/dotnet/commands/dotnet-workload/list/VisualStudioWorkloads.cs

```diff
@@ -119,6 +122,36 @@ internal static class VisualStudioWorkloads
             }
         }
 
+        /// <summary>
+        /// Writes install records for VS Workloads so we later install the packs via the CLI for workloads managed by VS.
+        /// This is to fix a bug where updating the manifests in the CLI will cause VS to also be told to use these newer workloads via the workload resolver.
+        /// ...  but these workloads don't have their corresponding packs installed as VS doesn't update its workloads as the CLI does.
+        /// </summary>
+        internal static void WriteSDKInstallRecordsForVSWorkloads(IInstaller workloadInstaller, IWorkloadResolver workloadResolver,
+            IEnumerable<WorkloadId> workloadsWithExistingInstallRecords, IReporter reporter)
+        {
+            // Do this check to avoid adding an unused & unnecessary method to FileBasedInstallers
+            if (OperatingSystem.IsWindows() && typeof(NetSdkMsiInstallerClient) == workloadInstaller.GetType())
+            {
+                InstalledWorkloadsCollection vsWorkloads = new();
+                GetInstalledWorkloads(workloadResolver, vsWorkloads);
+
+                // Remove VS workloads with an SDK installation record, as we've already created the records for them, and don't need to again.
+                var vsWorkloadsAsWorkloadIds = vsWorkloads.AsEnumerable().Select(w => new WorkloadId(w.Key));
+                var workloadsToWriteRecordsFor = vsWorkloadsAsWorkloadIds.Except(workloadsWithExistingInstallRecords);
+
+                if (workloadsToWriteRecordsFor.Any())
+                {
+                    reporter.WriteLine(
+                        string.Format(LocalizableStrings.WriteCLIRecordForVisualStudioWorkloadMessage,
+                        string.Join(", ", workloadsToWriteRecordsFor.Select(w => w.ToString()).ToArray()))
+                    );
+                }
+
+                ((NetSdkMsiInstallerClient)workloadInstaller).WriteWorkloadInstallRecords(workloadsToWriteRecordsFor);
```

Comment:
I don't think that we necessarily need to add a new method to `NetSdkMsiInstallerClient` and do an explicit cast here.  I think instead we could do something like:

```c#
foreach (var workload in workloadsToWriteRecordsFor)
{
    workloadInstaller.GetWorkloadInstallationRecordRepository().WriteWorkloadInstallationRecord(workload, new SdkFeatureBand(workloadResolver.GetSdkFeatureBand()));
}
```

We'd lose the additional log message, but that doesn't seem critical.